sopht/utils/restart_sim.py

```diff
@@ -0,0 +1,46 @@
+from typing import Type, Union
+import os
+import elastica as ea
+from sopht.simulator.flow.navier_stokes_flow_simulators import (
+    UnboundedNavierStokesFlowSimulator3D,
+    UnboundedNavierStokesFlowSimulator2D,
+)
+from sopht.simulator.flow.passive_transport_flow_simulators import (
+    PassiveTransportFlowSimulator,
+)
+from sopht.utils.io import IO
+
+
+def restart_simulation(
+    flow_sim: Union[
+        UnboundedNavierStokesFlowSimulator2D,
+        UnboundedNavierStokesFlowSimulator3D,
+        PassiveTransportFlowSimulator,
+    ],
```

Comment:
Maybe use the vertical bar `|` here? In Python 3.10 `|` seems to be preferred over `Union`. See https://peps.python.org/pep-0604/#add-a-new-operator-for-union-type1-type2 
```suggestion
    flow_sim: UnboundedNavierStokesFlowSimulator2D
    | UnboundedNavierStokesFlowSimulator3D
    | PassiveTransportFlowSimulator,
```